### PR TITLE
[Deposit Summary] Use default currency to be show by default

### LINKED ISSUE #10170: [Deposit Summary] Use default currency to be show by default

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #10170

### Description

The idea is here is to stop relying on the order of currencies that comes from the backend but to sort it, always showing default currency as the first. The PR also brings a small refactoring

### Testing instructions

* Use a store with multiple currencies
* Notice that the default currency is show first even when you change it on the we

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt [hunk 1]
```diff
@@ -98,6 +102,11 @@ class PaymentsHubDepositSummaryStateMapper @Inject constructor(
             "FAILED" -> PaymentsHubDepositSummaryState.Deposit.Status.FAILED
             else -> PaymentsHubDepositSummaryState.Deposit.Status.UNKNOWN
         }
+
+    sealed class Result {
+        data class Success(val overview: PaymentsHubDepositSummaryState.Overview) : Result()
+        object InvalidateInputDate : Result()
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt]: ❓ I'd like to confirm if the term `InvalidateInputDate` is used intentionally, or is it meant to be `InvalidInputDate`?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt [hunk 2]
```diff
@@ -31,31 +30,36 @@ class PaymentsHubDepositSummaryStateMapper @Inject constructor(
...
+                        pendingFundsAmount = pendingBalances.firstOrNull { it.currency == currency }?.amount ?: 0,
+                        pendingBalanceDepositsCount = pendingBalances.firstOrNull {
+                            it.currency == currency
+                        }?.depositsCount ?: 0,
+                        fundsAvailableInDays = overview.account?.depositsSchedule?.delayDays,
+                        fundsDepositInterval = overview.account.fundsAvailableIn(),
+                        nextDeposit = nextDeposits.firstOrNull { it.currency == currency }?.let { mapDeposit(it) },
+                        lastDeposit = lastPaidDeposits.firstOrNull { it.currency == currency }?.let { mapDeposit(it) }
+                    )
+                }.toSortedMap(
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt:59]: ❓ What do you think about writing unit tests for this particular sorting logic?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov. LGTM! I've left couple of minor comments that are non-blocking for the merge.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt]: Good catch - fixed here 43a4f6ac02cd0d35edcfe2deafbb9529e7ef5b38

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/depositsummary/PaymentsHubDepositSummaryStateMapper.kt:59]: That's here - 

`given overview with multiple currencies and different deposit statuses, when mapDepositOverviewToViewModelOverviews, then return Overview with correct statuses and sorted`

not dedicated test but it tests that

**Pull request merged**